filter/filter.go

```diff
@@ -1084,3 +1084,82 @@ func (c *Client) doPatchWithAuthHeaders(ctx context.Context, userAuthToken, serv
 	// do the request
 	return c.hcCli.Client.Do(ctx, req)
 }
+
+// CreateFilter creates a filter and returns the associated filterID and eTag
+func (c *Client) CreateFilter(ctx context.Context, userAuthToken, serviceAuthToken, downloadServiceToken, collectionID, datasetID, edition, version, populationType string, names []string) (filterID, eTag string, err error) {
```

Comment:
It would have been better to create a model that represents all the required data to create a filter.
This would avoid us having to refactor method signatures in the future. 

I have also realise that this is what we are doing for other methods.

Just a comment.